hack/validation/kubelet.sh

```diff
@@ -0,0 +1,9 @@
+# Kubelet evictionHard and evictionSoft must be a percent or a quanitiy value
+
+# Adding vaildation for nodeclaim
+yq eval '.spec.versions[0].schema.openAPIV3Schema.properties.spec.properties.kubelet.properties.evictionHard.additionalProperties.pattern = "^((\d{1,2}(\.\d{1,2})?|100(\.0{1,2})?)%||(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?)$"' -i pkg/apis/crds/karpenter.sh_nodeclaims.yaml 
```

Comment:
It might be helpful to add a comment here on exactly what this regex is doing so someone who is coming in, reading this for the first time, doesn't have to reason about what it's for